exchange/validation.go

```diff
@@ -0,0 +1,50 @@
+package exchange
+
+import (
+	"fmt"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type bidResponseWrapper struct {
+	adapterBids  *pbsOrtbSeatBid
+	adapterExtra *seatResponseExtra
+	bidder       openrtb_ext.BidderName
+}
+
+// validateBids will run some validation checks on the returned bids and excise any invalid bids
+func (brw *bidResponseWrapper) validateBids() (err []error) {
+	// Exit early if there is nothing to do.
+	if brw.adapterBids == nil || len(brw.adapterBids.bids) == 0 {
+		return
+	}
+	err = make([]error, 0, len(brw.adapterBids.bids))
+	validBids := make([]*pbsOrtbBid, 0, len(brw.adapterBids.bids))
+	for _, bid := range brw.adapterBids.bids {
+		if ok, berr := brw.validateBid(bid); ok {
+			validBids = append(validBids, bid)
+		} else {
+			err = append(err, berr)
+		}
+	}
+	if len(validBids) != len(brw.adapterBids.bids) {
+		// If all bids are valid, the two slices should be equal. Otherwise replace the list of bids with the valid bids.
+		brw.adapterBids.bids = validBids
+	}
+	return err
+}
+
+// validateBid will run the supplied bid through validation checks and return true if it passes, false otherwise.
+func (brw *bidResponseWrapper) validateBid(bid *pbsOrtbBid) (bool, error) {
```

Comment:
This could be a free function as it never uses `brw`.

Reply:
I figured since it would never be used for anything else, just keep it glued to brw.

Reply:
```
I figured since it would never be used for anything else, just keep it glued to brw.
```

It is true, but doesn't feel so right.. 

I believe both receiver arguments and function arguments are just arguments inside the function. So, it should be equivalent to `validateBid(brw *bidResponseWrapper, bid *pbsOrtbBid)`.  

Please correct me if I'm wrong. 😅 


If you would not do `validateBid(brw *bidResponseWrapper, bid *pbsOrtbBid)` when brw is not used inside, you would not want to do `func (brw *bidResponseWrapper) validateBid(bid *pbsOrtbBid)` either.

Reply:
I will free the function.